app/Repositories/PostRepository.php

```diff
@@ -77,6 +79,21 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
 
         $signup = Signup::find($signupId);
 
+        // Get the action_id either from the payload or the DB.
+        if (isset($data['action_id'])) {
+            $actionId = $data['action_id'];
+        } else {
+            $type = isset($data['type']) ? $data['type'] : 'photo';
+            $action = Action::where('campaign_id', $signup->campaign_id)->where('post_type', $type)->where('name', $data['action'])->first();
+
+            if (! $action) {
+                // Throw an exception that no action exists... should this be caught at an earlier point though?
+                throw new ModelNotFoundException('Action not found.');
+            } else {
+                $actionId = $action->id;
```

Comment:
I don't think this strictly needs to be in an `else` statement if you don't want it to be

Reply:
ah! Do you mean it might read better as a ternary?

Reply:
I _think_ you can remove the `else` entirely

Reply:
oh you're totally right! Removed :)